### ISSUE TITLE
How to train model without Y_cf

### ISSUE THREAD

**Issue by TianyiChen97** [ruoqi-liu/DSW#3]
Hi Ruoqi, we are trying to apply your model to our real data problem. I know your code is for simulation data, so we have to replace your data with our real data. Since for convenience, we would like to change as few codes as possible, so we decide to replace your A, X.all, X.Static and Y with our real data. but when we are planning so, we find out that there is Y_cf in your code. So i am wondering if we can get rid of Y_cf. or more a more general question, how can we import real data and apply your code to train the model. Thanks very much to any response or hint!

**Comment by ruoqi-liu** [comment]
> Hi Ruoqi, we are trying to apply your model to our real data problem. I know your code is for simulation data, so we have to replace your data with our real data. Since for convenience, we would like to change as few codes as possible, so we decide to replace your A, X.all, X.Static and Y with our real data. but when we are planning so, we find out that there is Y_cf in your code. So i am wondering if we can get rid of Y_cf. or more a more general question, how can we import real data and apply your code to train the model. Thanks very much to any response or hint!

Hi,

- In the training process: the loss is computed over Y (factual) as in line 85~87 in train_synthetic.py:
`outcome_loss = torch.mean(weights*(f_outcome_out - targets[:,0]) ** 2)`
`loss = ipw_loss + outcome_loss*0.1`
You may not use the predicted counterfactual outcomes from `ipw_outputs, f_outcome_out, cf_outcome_out, _ = model(x_inputs, x_static_inputs, fr_targets)`
- In the evaluation process, in line 225 of train_synthetic.py ("model_eval" function):
`pehe,ate = compute_pehe_ate(ipw_true, f_outcome_true, cf_outcome_true, f_outcome_outputs, cf_outcome_outputs)`
`mse = mean_squared_error(f_outcome_true,f_outcome_outputs) `
You can comment out the "compute_pehe_ate" since it is to evaluate the PEHE and ATE that using Y_cf. The "mean_squared_error" is to evaluate the MSE that only using Y.

**Comment by TianyiChen97** [comment]
Thanks very much. It works!
